src/context.cc

```diff
@@ -4816,63 +4829,117 @@ void Context::syncerActivity() {
             Poco::Thread::sleep(250);
         }
 
-        {
-            Poco::Mutex::ScopedLock lock(syncer_m_);
+        switch (state) {
+        case STARTUP: {
+            logger.log("Syncer bootup, will attempt to determine which protocol to use");
 
-            if (trigger_sync_) {
+            Poco::Mutex::ScopedLock lock(user_m_);
+            std::string api_token = user_->APIToken();
 
-                error err = pullAllUserData();
-                if (err != noError) {
-                    displayError(err);
+            HTTPRequest req;
+            req.host = urls::API();
+            req.relative_url = "/api/v9/me/flags";
+            req.basic_auth_username = api_token;
+            req.basic_auth_password = "api_token";
+
+            HTTPResponse resp = TogglClient::GetInstance().Get(req);
+
+            // don't do anything if the error is not noError, that's usually network error and this will try to connect every second
+            if (resp.err == noError) {
+                // if the server doesn't respond OK to this request, fall back to the legacy sync protocol
+                if (resp.status_code != 200) {
+                    logger.log("Syncer - Server didn't respond 200 to /me/flags, fallback to LEGACY");
+                    state = LEGACY;
+                }
+                else { // is OK - 200
+                    // otherwise, parse the response and see if the desktop_sync_client flag is there
+                    Json::Value root;
+                    Json::Reader reader;
+                    if (!reader.parse(resp.body, root)) {
+                        logger.log("Syncer - /me/flags response couldn't be parsed as JSON, fallback to LEGACY");
+                        state = LEGACY;
+                    }
+                    else {
+                        // there was a typo in the initial set of flags, use both variants to be sure
+                        if (root[kSyncStrategyLegacy1].isBool() && root[kSyncStrategyLegacy1].asBool())
+                            state = BATCHED;
+                        else if (root[kSyncStrategyLegacy2].isBool() && root[kSyncStrategyLegacy2].asBool())
+                            state = BATCHED;
+                        else
+                            state = LEGACY;
+                        logger.log("Syncer - Syncing protocol was selected: ", (state == BATCHED ? "BATCHED" : "LEGACY"));
+                    }
                 }
```

Comment:
I suggest having an `else` statement here to rollback to `LEGACY` mode if the endpoint gets errors.

It would be:
```C++
else {
    state = LEGACY;
}
```
Thus, in the worst scenario that `/api/v9/me/flags` is down, the app still does Legacy Sync properly

Reply:
If the endpoint is down, the whole API is very likely to be down as well, resulting in not getting any data from/to the API anyway.

Reply:
That also accounts for cases like when you don't have internet connection at all, too.

Reply:
That makes sense 👍